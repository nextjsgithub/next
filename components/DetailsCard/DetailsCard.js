import React from 'react';
import MainButton from '../Buttons/MainButton';

const DetailsCard = ({ onClick, showDetails, edges }) => {
  let commitInfo;
  if (edges[showDetails].node.object) {
    commitInfo = edges[showDetails].node.object.history.edges.map(
      (commit, i) => {
        return (
          <div key={i} className="DetailsCard__textCommitsWrapper">
            <div>
              <span className="DetailsCard__textKey">Author login:</span>
              <span className="DetailsCard__textAuthor">
                {commit.node.author.name}
              </span>
            </div>
            <div className="DetailsCard__textKey">Commit message:</div>
            <div className="DetailsCard__commitText">{commit.node.message}</div>
          </div>
        );
      }
    );
  } else {
    commitInfo = [<div key={1}>No commits / no master branch</div>];
  }

  return (
    <div className="DetailsCard__wrapper">
      <div className="DetailsCard__textPairWrapper">
        <span className="DetailsCard__textKey">Project name:</span>
        <span className="DetailsCard__textValue">
          {edges[showDetails].node.name}
        </span>
      </div>
      <div className="DetailsCard__textPairWrapper">
        <span className="DetailsCard__textKey">Owner:</span>
        <span className="DetailsCard__textValue">
          {edges[showDetails].node.owner.login}
        </span>
      </div>
      <div className="DetailsCard__textDescriptionWrapper">
        <span className="DetailsCard__textKey">Link to clone repository:</span>
        <span className="DetailsCard__textValueUrl">
          {edges[showDetails].node.url}.git
        </span>
      </div>
      <div className="DetailsCard__textDescriptionWrapper">
        <span className="DetailsCard__textKey">Description:</span>
        <span className="DetailsCard__textDescription">
          {edges[showDetails].node.description}
        </span>
      </div>
      <div className="DetailsCard__textDescriptionWrapper">
        <span className="DetailsCard__textTitle">Last commits:</span>
        <span className="DetailsCard__textDescription">{commitInfo}</span>
      </div>
      <div className="DetailsCard__btnWrapper">
        <MainButton type="button" onClick={() => onClick('')}>
          Back to list
        </MainButton>
      </div>
    </div>
  );
};

export default DetailsCard;
