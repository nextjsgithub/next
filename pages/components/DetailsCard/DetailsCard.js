import React from 'react';
import MainButton from '../Buttons/MainButton'

const DetailsCard = ({ onClick, showDetails, edges }) => {
  return (
    <div key={showDetails} style={{ display: 'flex', flexDirection: 'column' }}>
      <div>
        {showDetails + 1}. Project name: {edges[showDetails].node.name}
      </div>
      <div style={{ padding: '0 0 0 20px' }}>
        Owner: {edges[showDetails].node.owner.login}
      </div>
      <div style={{ padding: '0 0 0 20px' }}>
        Link:{' '}
        <MainButton type="button" onClick={() => onClick(false)}>
          Back to list
        </MainButton>
      </div>
    </div>
  );
};

export default DetailsCard;
