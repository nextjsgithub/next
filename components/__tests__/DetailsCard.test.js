import React from 'react';
import { render } from 'enzyme';
import DetailsCard from '../DetailsCard/DetailsCard';

let wrapper;

const edges = [JSON.parse(`{
  "__typename": "SearchResultItemEdge",
  "node": {
    "__typename": "Repository",
    "id": "MDEwOlJlcG9zaXRvcnkxMzcwNzg0ODc=",
    "name": "vue-next",
    "description": "ðŸ–– Vue.js is a progressive, incrementally-adoptable JavaScript framework for building UI on the web.",
    "url": "https://github.com/vuejs/vue-next",
    "owner": {
      "__typename": "Organization",
      "login": "vuejs"
    },
    "object": {
      "__typename": "Commit",
      "id": "MDY6Q29tbWl0MTM3MDc4NDg3OjQ3ZGE5MjE0NmM5ZmIzZmE2YjFlMjUwZTA2NGNhNDliNzRkODE1ZTQ=",
      "history": {
        "__typename": "CommitHistoryConnection",
        "edges": [
          {
            "__typename": "CommitEdge",
            "node": {
              "__typename": "Commit",
              "id": "MDY6Q29tbWl0MTM3MDc4NDg3OjQ3ZGE5MjE0NmM5ZmIzZmE2YjFlMjUwZTA2NGNhNDliNzRkODE1ZTQ=",
              "author": {
                "__typename": "GitActor",
                "name": "HcySunYang"
              },
              "message": "fix(compiler-core): fix whitespace management for slots with whitespace: 'preserve' (#3767)\\n\\nfix #3766"
            }
          },
          {
            "__typename": "CommitEdge",
            "node": {
              "__typename": "Commit",
              "id": "MDY6Q29tbWl0MTM3MDc4NDg3OmYzZDMwMzYzZWMzM2NlNmI2MjBmNjdlNGNiOGRiZjhiODZjZWY1ODM=",
              "author": {
                "__typename": "GitActor",
                "name": "Zen"
              },
              "message": "chore: bump api-extractor from 7.12.1 to 7.15.1 (#3745)"
            }
          },
          {
            "__typename": "CommitEdge",
            "node": {
              "__typename": "Commit",
              "id": "MDY6Q29tbWl0MTM3MDc4NDg3OjQ5ZmQxNjk2YzkzMTEwODNhN2E4N2RhZmU5ZDAwOTg4MzI4NGVmM2M=",
              "author": {
                "__typename": "GitActor",
                "name": "Yunfei He"
              },
              "message": "chore(types): remove unnecessary type assertion (#3724)"
            }
          },
          {
            "__typename": "CommitEdge",
            "node": {
              "__typename": "Commit",
              "id": "MDY6Q29tbWl0MTM3MDc4NDg3OjliMTYwYjk0MDU1NWFiYjZiNmNlNzIyZmRkYmQ5NjQ5ZWUxOTZmN2I=",
              "author": {
                "__typename": "GitActor",
                "name": "GU Yiling"
              },
              "message": "fix(types): declared prop keys should always exist in \`props\` argument (#3726)"
            }
          }
        ]
      }
    }
  }
}`)]

beforeEach(() => {
  wrapper = render(<DetailsCard edges={edges} showDetails={0} />);
});

describe('<DetailsCard /> rendering', () => {
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
