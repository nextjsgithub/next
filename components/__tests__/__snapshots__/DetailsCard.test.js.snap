// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetailsCard /> rendering renders correctly 1`] = `
<div
  class="DetailsCard__wrapper"
>
  <div
    class="DetailsCard__textPairWrapper"
  >
    <span
      class="DetailsCard__textKey"
    >
      Project name:
    </span>
    <span
      class="DetailsCard__textValue"
    >
      vue-next
    </span>
  </div>
  <div
    class="DetailsCard__textPairWrapper"
  >
    <span
      class="DetailsCard__textKey"
    >
      Owner:
    </span>
    <span
      class="DetailsCard__textValue"
    >
      vuejs
    </span>
  </div>
  <div
    class="DetailsCard__textDescriptionWrapper"
  >
    <span
      class="DetailsCard__textKey"
    >
      Link to clone repository:
    </span>
    <span
      class="DetailsCard__textValueUrl"
    >
      https://github.com/vuejs/vue-next.git
    </span>
  </div>
  <div
    class="DetailsCard__textDescriptionWrapper"
  >
    <span
      class="DetailsCard__textKey"
    >
      Description:
    </span>
    <span
      class="DetailsCard__textDescription"
    >
      ðŸ–– Vue.js is a progressive, incrementally-adoptable JavaScript framework for building UI on the web.
    </span>
  </div>
  <div
    class="DetailsCard__textDescriptionWrapper"
  >
    <span
      class="DetailsCard__textTitle"
    >
      Last commits:
    </span>
    <span
      class="DetailsCard__textDescription"
    >
      <div
        class="DetailsCard__textCommitsWrapper"
      >
        <div>
          <span
            class="DetailsCard__textKey"
          >
            Author login:
          </span>
          <span
            class="DetailsCard__textAuthor"
          >
            HcySunYang
          </span>
        </div>
        <div
          class="DetailsCard__textKey"
        >
          Commit message:
        </div>
        <div
          class="DetailsCard__commitText"
        >
          fix(compiler-core): fix whitespace management for slots with whitespace: 'preserve' (#3767)

fix #3766
        </div>
      </div>
      <div
        class="DetailsCard__textCommitsWrapper"
      >
        <div>
          <span
            class="DetailsCard__textKey"
          >
            Author login:
          </span>
          <span
            class="DetailsCard__textAuthor"
          >
            Zen
          </span>
        </div>
        <div
          class="DetailsCard__textKey"
        >
          Commit message:
        </div>
        <div
          class="DetailsCard__commitText"
        >
          chore: bump api-extractor from 7.12.1 to 7.15.1 (#3745)
        </div>
      </div>
      <div
        class="DetailsCard__textCommitsWrapper"
      >
        <div>
          <span
            class="DetailsCard__textKey"
          >
            Author login:
          </span>
          <span
            class="DetailsCard__textAuthor"
          >
            Yunfei He
          </span>
        </div>
        <div
          class="DetailsCard__textKey"
        >
          Commit message:
        </div>
        <div
          class="DetailsCard__commitText"
        >
          chore(types): remove unnecessary type assertion (#3724)
        </div>
      </div>
      <div
        class="DetailsCard__textCommitsWrapper"
      >
        <div>
          <span
            class="DetailsCard__textKey"
          >
            Author login:
          </span>
          <span
            class="DetailsCard__textAuthor"
          >
            GU Yiling
          </span>
        </div>
        <div
          class="DetailsCard__textKey"
        >
          Commit message:
        </div>
        <div
          class="DetailsCard__commitText"
        >
          fix(types): declared prop keys should always exist in \`props\` argument (#3726)
        </div>
      </div>
    </span>
  </div>
  <div
    class="DetailsCard__btnWrapper"
  >
    <button
      class="MainButton__Btn"
      type="button"
    >
      Back to list
    </button>
  </div>
</div>
`;
